include/soralog/util.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SORALOG_UTIL
+#define SORALOG_UTIL
+
+#include <pthread.h>
+#include <array>
+#include <string>
+
+namespace soralog::util {
+
+  inline void setThreadName(std::string_view name) {
+    std::array<char, 16> buff{};
+    memcpy(buff.data(), name.data(), std::min<size_t>(name.size(), 15));
+#if defined(__linux__)
+    pthread_setname_np(pthread_self(), buff.data());
+#elif defined(__APPLE__)
+    pthread_setname_np(buff.data());
+#endif
+  }
+
+  inline void getThreadName(std::array<char, 16> &name) {
+    static thread_local std::array<char, 16> thr_name{};
+    static thread_local bool initialized = [&] {
+#if defined(__linux__)
```

Comment:
If this isn't working for Windows, I think it would be a good idea to explicitly emit an #error on windows.

Reply:
Also, aren't branches for linux and apple the same?

Reply:
Fixed